data/human/ships.txt

```diff
@@ -332,22 +332,32 @@ ship "Bactrian"
 		"X4700 Ion Thruster"
 		"X5200 Ion Steering"
 		"Hyperdrive"
-		
-	engine -18 230
-	engine 18 230
-	gun -15 -226 "Torpedo Launcher"
-	gun 15 -226 "Torpedo Launcher"
-	gun -40 -133 "Sidewinder Missile Launcher"
-	gun -45 -128 "Sidewinder Missile Launcher"
-	turret -22 -148 "Heavy Laser Turret"
-	turret 26 -80 "Heavy Anti-Missile Turret"
-	turret -41 -20 "Heavy Laser Turret"
-	turret 32 -7 "Heavy Laser Turret"
-	turret 53 82 "Heavy Laser Turret"
-	turret -37 186 "Heavy Anti-Missile Turret"
-	bay "Fighter" -38 -26
-	bay "Fighter" 35 36
-	bay "Fighter" -47 97
+
+	engine 19 246.5
+		zoom 1
+		angle 0
+		under
+	engine -19 246.5
+		zoom 1
+		angle 0
+		under	
+	gun -67.5 65 "Sidewinder Missile Launcher"
+		angle 305
+	gun -62.5 56.5 "Sidewinder Missile Launcher"
+		angle 305
+	gun 67 -126 "Torpedo Launcher"
+		angle 40
+	gun 74 -120 "Torpedo Launcher"
+		angle 40
```

Comment:
I'm not a fan of using angled gun ports here. It's an idea that I've discussed in the past for alien ships, but I'd rather not have angled gun ports on any human ships, reason being to keep the more interesting mechanics for later in the game instead of showing off everything in human space.

Reply:
This would make considerable sense if we were discussing, say, putting slowing weapons on a Bactrian. However, this is not what we are discussing. We are discussing gun-angles, which should not be considered a "special feature", but rather a standard element of ship design to make our current often-overly-similar ship designs more varied.

In the past I have highlighted two facets of the Bactrian that have generally been contradictory:
1. As a *city*-ship, as well as being a ship of its size, guns are inappropriate armament. It should *never* be spinning around fast enough for guns to be a meaningful weapon; even if our current outfitting system means that it frequently happens.
2. The Bactrian's lore emphasizes its role as being a ship used to venture into unsettled territory to establish and supply colonies. Its defensive capabilities are justified as being there to protect it from ambushes while enroute.
3. The Bactrian's stock loadout for the guns revolves around homing projectiles. As a general principle, homing projectiles are ideal ordinance for a heavy ship looking to take down agile attackers. 

Now, ambushes, by general definition, happen from inopportune (to the defender) angles: Coming in from hyperspace behind them, around planets or large asteroids to the sides, etc. Virtually by definition ambushes and raids are things that come from the sides, while *blockades* are opposition from the front. Nowhere is the Bactrian described as being a blockade runner. As such, virtually all the combat it is expected to be engaged in involves combatants coming at it from the sides or rear. Further more, given the size of the ship, it is virtually guaranteed that it is facing ships that are considerably more agile than it.

There is also the problem that, in its present vanilla configuration, Bactrians are overly good in my opinion at being warships. This is, in large part, due to the number of guns that can be fitted with heavy lasers or other high powered guns. I have in the past PR'd changes to reduce the number of guns, which have been denied. I'm sure there were a variety of reasons, but one of the ones I recall was the fact that having a large number of missiles weapons was, in fact, both perfectly reasonable and in keeping with the lore for this ship.

In this PR we have something that works to strengthen and satisfy both these elements:
3. Having angled gun placements starts off projectiles at a 45 from the Bactrian's forward axis, which gives projectiles a good head start in their turn to strike at ships that are anywhere but in front of it.
2. Given the original design expecting to counter ambushes, having the ports setup this way maximizes the offensive potential of the ship against ambushers and pursuers.
1. This means that the Bactrian doesn't need to spin nearly as much to bring its weapons to bear. As such, it can focus on getting to its destination or using that outfitting space for other things.
0. It also reduces the suitability of using these gunports for guns. I'm sure some people will still find creative flight profiles that make good use of them for enabling a strafing style attack pattern;  but for the most part people will simply skip putting guns in them and utilize homing projectiles as intended. This reinforces the previous three points.

In other words, having these guns angled as they are makes them more in line with their lore, more effective in their intended usage as homing projectile launchers, and less effective in their non-intended use for guns., which in turn reduces the need for Bactrians to be able to turn quickly since none of their weapons depend on it.

In my view, this makes this change a win for each of our perspectives, a win for the Bactrian's lore, and a win for demonstrating to our community what the game engine can actually handle. Given that even most of the people on Discord haven't yet even realized that angled gun are at thing. What better place to showcase it than on an eminently sought after ship like this?

Reply:
The discord poll is less than 24h old, so I'm not sure everyone's gotten a chance at it. That being said, the vote tally is currently 339 to 31 in favor of the angled gun ports. While obviously that doesn't change the fact that the Devs get veto, I'd say this is a prime situation where we can follow the Kestrel (guns) experiment: Keep the angled guns in for the Bactrian for now, and see how things go. In a few months after everyone's had a chance to actually *experience* how they play in the course of regular gameplay, we re-evaluate. 

In any case, my interpretation of the discussion is that the community is firmly on the side of "gun angle is a basic feature of ship design, not something fancy for future factions" line of thought. Not unanimous, but pretty strong.

Reply:
The poll has 4 points supporting the angled gun ports, written by the one who made the poll and is in favor of them, and a very short summary of points against it that were not even written out or developed in this PR yet, instead of letting the person against the angle ports write their own points and have those listed out just as the arguments for having these angled ports are.

Whether that'd change the results of the poll or not, it's still a very bad practice to create a poll that'll of course be biased when one presents several arguments for one side and a diminutive summary of the other.

Reply:
Valid point, but at 10:1 in favour, it's probably negligible.

Reply:
Debatable.

Reply:
There was one point given against it. Still is only one point against it. I would assume that if someone is going to list only a single reason to oppose something, I would expect them too pick their strongest point. In any case, the community seems pretty unimpressed with the idea of saving gun angles as an "advanced" thing for higher tech races/factions.

Reply:
So, how do we resolve this? Ask again with a neutral text? Disregard the poll in this discussion?

Reply:
It's not a fair poll, so I don't see why it should be taken into account. I expect a fair poll to still favor the angled gun ports, but haven't had a fair poll to know.

Reply:
And how do you define a fair poll? Limit the pro side to a single argument? I prefer to present all the arguments, and if one side has more than the other, then that is how the cookie crumbles. The question is what do we think makes the game better, not how to ensure both sides get an even 50% of the text in the poll.

Reply:
I never said we should limit the arguments to a specific number, I never said both sides needed to have the same amount of arguments. I stated the fact that the poll is unfair because one side has its arguments there and the other doesn't, it has a tiny summary of what may be just one of that side's arguments.
You didn't give the other side time to give their arguments, you just rushed a poll out before the other side's arguments were given.
Caring for what makes the game better involves listening to what the other side has to say, and if we're relying on the community to decide that then we should wait for the other side to say what it has to say before summarizing the other side in one line while we give ourselves the luxury of presenting our arguments in full, in the poll that we created.

Reply:
That is assuming there is anything else. Amazinite is Dev. Stuff can and does get shutdown based on his opinion alone. Since nothing more substantial was forthcoming, I made the poll based on the complete available arguments. I believe, and seems the community largely agrees with me, that his stated opinion in this case is insufficient justification.

The fact that nothing else has been said also suggests this.

Reply:
And that's assuming there is nothing else.

Reply:
Looking through the poll, I still think it's biased. One could have at the very least tried to keep the 2 arguments to equal size, by taking derpy's entire comment against a shortened version of Zitchas's.
Derpy's point also is not invalid. We know that people can get swayed a lot by first impression, and I did, on first impression, dismiss derpy's point. which I now see as perfectly valid. the 10:1 ratio *could* have been caused by the arguments presented being very lopsided.

Reply:
This reminds me of the climate change "debate." 99.9% of climate scientists firmly endorse the science of climate change, and have for decades. But many news outlets tried to present it fairly and gave equal coverage to both "pro" and "con" information on climate change. Many people thus erroneously believed it was actually a disputed thing. 

If one side has many arguments for it, and the other side has one argument, all that means is that one side doesn't have much going for it. It doesn't make things less fair.

That being said, definitely, my framing of the poll wasn't done in the most impartial tone. But limiting it to one vs one argument is artificially inflating a position that, as of yet, still has only a single argument for it.

Reply:
Just because I provided a single point against using angled gun ports on the Bactrian doesn't mean that I only have a single point. You're correct in stating that given only a single point, someone is going to put their strongest point forward, but I'd say it's also important to note that how strong or persuasive a point is is going to change from person to person.

I provided the point of gating novel and unique mechanics behind later content because to me that is a very important point. It is me coming at this from a game design perspective, looking at the game as a whole and how we spread out content and ideas. I imagine though and it's perhaps evident that that is not the most persuasive point generally, at least when stacked up against a multitude of points going the other way, and I fully understand why that's the case as most people likely aren't coming at this from an overall game design perspective. I imagine for most people they're looking at it more from a "Oh that sounds cool, I like it," type of thing; very surface level. Now that isn't a jab, that's just the reality that most people aren't spending a significant amount of time formulating their thoughts on the finer aspects of game design for some random open source indie game they play from time to time.

So given that, I'm not providing my most persuasive point(s) for a general audience when reviewing a PR, I'm going to provide the point that I feel the strongest about, as reviews are intended for the creator of the PR and the point that I feel the most strongly about is what is the biggest hurdle for getting a PR merged and therefore what the creator of the PR should focus on. I could provide every point behind why I disagree with something when I review, but honestly I already spend a rather significant amount of time formulating reviews as is, particularly when there's some sort of strong disagreement involved and I want to carefully word my review (I'd estimate that the review for this PR took maybe close to an hour to write [and this post has taken even longer]; granted that was with me throwing together images in GIMP to show examples of my point and I was speaking in Discord at the same time, but it's still a large chunk of time for even what looks like a smaller review), and listing every point I have just becomes a waste of time for the PR creator to read through given that just my strongest held point against something should be enough, at least for an initial review; if the PR creator isn't convinced by my strongest held point then I can provide more after that, but from what I understand, and correct me if I'm wrong, but @Saugia doesn't have a particularly strong opinion on the state of the gun ports one way or the other, so listing every point I have would just be an exercise in convincing everyone else.

Given the storm that's been kicked up around this though I suppose I can provide more points, although I will admit that a lot of these have already been described by others on Discord, so it's not like the arguments weren't out there. In no particular order:
1) Given that what we are doing is suggesting a change for existing content, I'd say that the burden is on the people advocating for the change to convince me why the change to existing content is good. If this PR were adding a new ship to the game that used angled ports intended only for missiles then I would take less issue with it; if it were a human ship I'd still not like it, but if it were an alien ship of some sort, particularly outside of the Hai, then I might be perfectly fine with it.
2) Angled gun ports *restrict* the customizability of the Bactrian, which I do not think we should do. By angling the gun ports, the only viable guns that could be installed on the Bactrian are weapons with homing. Excluding the smaller human missile launchers since they basically just inflate the numbers (there are five of them if you want to include them), there are currently 43 weapons that a Bactrian could install in its gun ports. By angling the ports, you're restricting that to 14. (For this reason, I'd find it difficult to take any existing ship and convert any forward facing gun ports into angled gun ports, since you're removing options from the player, not adding them.)
3) I do not see the stock fitting of a ship as being justification for changing it such that only fittings like the stock fitting are viable. That is to say, just because the Bactrian comes stock with missiles doesn't mean that the Bactrian should only ever be capable of using missiles in its gun ports.
4) There have been suggestions that these angled gun ports could still be used with normal guns, which I'd contest simply isn't the case. I personally play with the auto aim setting on, and I imagine that most people do as well. What the auto aim does is if your ship is facing close enough to a target, the game automatically turns your ship to face the target, but should your gun ports be angled, this means that the game is aiming your gun ports off of the target, and given that the suggested angles aren't particularly far away from facing forward, this will occur often. So should you want to use normal guns in an angled port, either you need to have auto aim off or you need to not target the ship you are attacking, which means you're depriving yourself of information about your target ship (unless you have the overlay active, but I don't use that, and at that point you're talking about basically needing to use specific settings to make a loadout viable, which I don't think is a situation that we should be creating), and in both cases you need to manually aim your guns, which leads to human inaccuracy that's going to hurt your combat capabilities. So angled gun ports with normal guns simply can't be used normally like forward facing gun ports with normal guns.
5) Another suggestion for how angled ports could be used with normal guns is by doing "drive-bys" where you fly past a ship while firing your guns, but this simply isn't an effective tactic given that the time that your weapons will actually be on target will be so miniscule compared to if you simply had your guns facing forward and were facing your target head on or were using turrets that actually track a target as you fly past it. You'd need very particular types of weapons with high burst damage to actually be effective with this drive-by tactic, but those types of weapons simply don't exist, at least not in large numbers; you might be able to get away with it with a Nullifier or Detainer, but that's moreso because of the special effects those weapons have, and that's only two more weapons that could viably be used in angled ports. So insofar as we lack such weapons, particularly in human space, and no I wouldn't be interested in creating such weapons just to justify this one ship having angled ports, I don't find the drive-by tactic idea convincing as for how angled gun ports wouldn't hurt customization.
6) Going off of the auto aim idea that auto aim actually forces these gun ports into bad positions for using normal guns, the same is actually also true of missiles. While it is true that should a target be behind the Bactrian, angling the gun ports outward somewhat would allow the missiles to more quickly reach their target, but the fact is that ships face the targets they are firing upon, so by angling the gun ports outward you're actually increasing the travel time for any missiles to reach their targets under normal circumstances while only providing a slim benefit to the initial part of combat where a ship may not be facing its target.
7) I simply am not convinced that this focus on "Well it's a city-ship" means that we should be gimping the Bactrian's combat capabilities, either in reducing its customization or simply creating suboptimal firing angles for typical combat scenarios. The fact that it is a city-ship actually means, as the description states, that it is *designed* to be a *capable* warship, which it certainly is now. But the fact that it isn't a *dedicated* warship should not mean that we nerf its combat capabilities, especially seeing as a dedicated heavy warship is fully capable of taking on a Bactrian in combat; the Cruiser, Carrier, Dreadnought, and Kestrel as the other high-end human heavy warships are all able to take on a Bactrian, so among the high-end human heavy warships the Bactrian is *already at the bottom of the list as is in terms of combat*, and even low-end heavy warships like the Vanguard, Falcon, and Leviathan can take on a Bactrian given the right fittings, although granted that's in a player vs AI scenario as opposed to a head on brawl like the AI likes to do, in which case the winner is always going to be the ship with the bigger guns and higher HP, so it's a no-brainer that the Bactrian would beat a smaller warship.

Reply:
Thank you for the detailed response. I really appreciate it! I won't response immediately, but I'll think about it and respond.

Reply:
For the sake of everyone's sanity, and to work on getting the main feature of this PR, the models in, I have elected to not use angled gun ports (which was mentioned in this PR at a previous time if you're from the future.) If this is still desired, I strongly encourage those who wish it to do so in a separate PR, and tag this one if you so desire.

If you could wait for Zitchas's response, Derpy, before finalizing anything, that would be great, as I to want to hear his opinion as well. But once we're past this, I don't want to dwell on it that much longer.